src/bootstrap/src/core/config/config.rs

```diff
@@ -2407,18 +2427,26 @@ impl Config {
         };
 
         // Handle running from a directory other than the top level
-        let top_level = output(self.git().args(["rev-parse", "--show-toplevel"]));
-        let top_level = top_level.trim_end();
-        let compiler = format!("{top_level}/compiler/");
-        let library = format!("{top_level}/library/");
+        let compiler_path = format!("{}", self.src.join("compiler/").display());
+        let library_path = format!("{}", self.src.join("library/").display());
+
+        // If we are running in a CI environment, there are two cases that need to be handled:
+        //
+        //   1: When compiler sources have not changed, we need to use the previous commit (HEAD^) because
+        //   the current one does not have the artifacts built.
```

Comment:
I don't think this logic makes sense. If the compiler sources have not changed, then the preexisting selection process should be sufficient: we already would select an older commit than the one being built as the source, right?

Reply:
True, this is still wrong.

> then the preexisting selection process should be sufficient: we already would select an older commit than the one being built as the source, right?

It depends. If we are running a CI job on the master branch (which I am not sure), where the last commit was pushed by bors, then bootstrap will pick the last commit. But in that case, since it was pushed by bors I believe it will have the artifacts already built (also not sure with this), so it should be fine to use it with `download-rustc`. If you confirm that this is true, I can revert this change as the default behaviour will handle everything correctly.

Reply:
There's not really any CI on master but if there was it indeed should just work, the same way that a local fresh checkout should always work.

Reply:
> If you confirm that this is true, I can revert this change as the default behaviour will handle everything correctly.

Reverted.